test/LANDRegistry.js

```diff
@@ -1098,4 +1099,176 @@ contract('LANDRegistry', accounts => {
+        decayed.should.be.false
+      })
+
+
+      it('should return false if no gracePeriod set', async function() {
+        const assetId = await land.encodeTokenId(0, 1)
+        const decayed = await land.hasDecayed(assetId)
+        decayed.should.be.false
+      })
+
+      it('should return false if gracePeriod is bigger than today (one month from now)', async function() {
+        const assetId = await land.encodeTokenId(0, 1)
+        const date = new Date()
+        const oneMonth = new Date(date.getFullYear(), date.getMonth() + 1, 0)
+          .getDate() * 24 * 60 * 60 // In seconds
+
+        await land.setGracePeriod(oneMonth, sentByCreator)
+        const decayed = await land.hasDecayed(assetId)
+        decayed.should.be.false
+      })
+    })
+
+    describe('ping', function() {
+      it('should refresh latestPing if pinged by owner', async function() {
+        const latestPingBefore = await land.latestPing(user)
+        await land.pingMyself(sentByUser)
+        const latestPingAfter = await land.latestPing(user)
+        latestPingAfter.should.be.bignumber.equal(Math.trunc(new Date().getTime() / 1000))
+        ;(latestPingBefore < latestPingAfter).should.be.true
+      })
+
+      it('should refresh latestPing if pinged by updateManager', async function() {
+        const latestPingBefore = await land.latestPing(user)
+        await land.setUpdateManager(user, anotherUser, true, sentByUser)
+        await land.ping(user, sentByAnotherUser)
+        const latestPingAfter = await land.latestPing(user)
+        latestPingAfter.should.be.bignumber.equal(Math.trunc(new Date().getTime() / 1000))
+        ;(latestPingBefore < latestPingAfter).should.be.true
+      })
+
+      it('should refresh latestPing if pinged by approvedForAll', async function() {
+        const latestPingBefore = await land.latestPing(user)
+        await land.setApprovalForAll(anotherUser, true, sentByUser)
+        await land.ping(user, sentByAnotherUser)
+        const latestPingAfter = await land.latestPing(user)
+        latestPingAfter.should.be.bignumber.equal(Math.trunc(new Date().getTime() / 1000))
+        ;(latestPingBefore < latestPingAfter).should.be.true
+      })
+
+      it('should emit Ping event when ping', async function() {
+        const { logs } = await land.pingMyself(sentByUser)
+        const log = logs[0]
+        log.event.should.be.eq('Ping')
+        log.args._caller.should.be.bignumber.equal(user)
+        log.args._holder.should.be.equal(user)
+      })
+
+      it('should emit Ping event when ping by other', async function() {
+        await land.setApprovalForAll(anotherUser, true, sentByUser)
+        const { logs } = await land.ping(user, sentByAnotherUser)
+        const log = logs[0]
+        log.event.should.be.eq('Ping')
+        log.args._caller.should.be.bignumber.equal(anotherUser)
+        log.args._holder.should.be.equal(user)
+      })
+
+      it('reverts if ping non-authorized address', async function() {
```

Comment:
`reverts if ping non-authorized address` by `reverts if trying to ping by a non-authorized address`